### PR TITLE
FINERACT-1719: Introduce Lombok to mix module

### PR DESCRIPTION
**Pull request by hmg65**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/mix/data/MixTaxonomyData.java [hunk 1]
```diff
@@ -26,42 +33,11 @@ public class MixTaxonomyData {
...
-
-    public String getDimension() {
-        return this.dimension;
-    }
-
-    public Integer getType() {
-        return this.type;
-    }
-
-    public boolean isPortfolio() {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/mix/data/MixTaxonomyData.java:64]: Why remove this?

File: fineract-provider/src/main/java/org/apache/fineract/mix/service/XBRLBuilder.java [hunk 2]
```diff
@@ -145,7 +145,7 @@ Element addTaxonomy(final Element rootElement, final MixTaxonomyData taxonomy, f
     }
 
     private String getUnitRef(final MixTaxonomyData tx) {
-        return tx.isPortfolio() ? UNITID_PURE : UNITID_CUR;
+        return tx.getType() == 5 ? UNITID_PURE : UNITID_CUR;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/mix/service/XBRLBuilder.java]: Let's revert this line.

File: fineract-provider/src/main/java/org/apache/fineract/mix/data/ContextData.java [hunk 3]
```diff
@@ -18,29 +18,18 @@
...
-    private final String dimension;
-    private final Integer periodType;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
 
-    public ContextData(final String dimensionType, final String dimension, final Integer taxonomyType) {
-        this.dimensionType = dimensionType;
-        this.dimension = dimension;
-        this.periodType = taxonomyType.equals(MixTaxonomyData.BALANCESHEET) || taxonomyType.equals(MixTaxonomyData.PORTFOLIO) ? 0 : 1;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/mix/data/ContextData.java:30]: This logic is missing where the object gets instantiated.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/mix/service/XBRLBuilder.java (hunk 2 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/mix/service/XBRLBuilder.java]: done

File: fineract-provider/src/main/java/org/apache/fineract/mix/data/ContextData.java (hunk 3 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/mix/data/ContextData.java:30]: corrected

**Review by galovics — APPROVED** [review]

**Pull request merged**